docs/migration-recipe.md

```diff
@@ -0,0 +1,106 @@
+
+To view haproxy status: 
+
+-vagrant up the salt-master, loadbalancer, and host in question (`vagrant up`)
+
+-`vagrant ssh-config salt-master loadbalancer >> vagrant-ssh Ssh -L 4646:127:0.0.1:464 -F vagrant-ssh loadbalancer`
+
+-open loacalhost:4646/haproxy?stats in browser
+
+2.  Edit pillar data for roles.sls to include both old and new hostnames (ex. hostname*)
+
+### Migrate the host:
+
+#### Update Salt Master with latest config including prep from above
+
+1.  ssh into the salt-master server `ssh salt-master`
+
+2.  Navigate to srv/psf-salt `cd srv/psf-salt`
+
+3.  Run `git pull`
+
+4.  Run `sudo salt-call state.highstate`
+
+#### Ensure new configuration doesn't impact host being migrated
+
+1.  ssh into the host in question `ssh hostname`
+
+2.  Navigate to srv/psf-salt `cd srv/psf-salt`
+
+3.  Run `git pull`
+
+4.  Run `sudo salt-call state.highstate`
+
+#### Create new host:
+
+1.  Start a new droplet  in digital ocean, and check resources being used on old host to see if we are over or under spending on resources 
+
+2.  Create a new droplet with new version of ubuntu, appropriate resources,  and name it according to hostname + 2004
+
+#### Provision new host for migration:
+
+1.  Add Salt repositories for our current target version (add the apt-repo and install salt-minion package)
+
+-  `RUN wget --quiet -O /usr/share/keyrings/salt-archive-keyring.gpg https://repo.saltproject.io/py3/ubuntu/20.04/$(dpkg --print-architecture)/3004/salt-archive-keyring.gpg`
+
+-  `RUN echo "deb [signed-by=/usr/share/keyrings/salt-archive-keyring.gpg arch=$(dpkg --print-architecture)] https://repo.saltproject.io/py3/ubuntu/20.04/$(dpkg --print-architecture)/3004 focal main" > /etc/apt/sources.list.d/salt.list`
+
+2.  Configure the salt-minion. On new host, run the command
+
+- `RUN apt-get update -y && apt-get install -y --no-install-recommends salt-minion`
+
+- On the original host, look through `etc/salt/minion.d*` to set up salt-minion configuration files to match on new host 
+
+3.  Run `sudo salt-call state.highstate`
+
+#### Stop services on old host:
+
+1.  ssh into old host ( `ssh old-hostname` )
+
+2.  stop cron jobs, `sudo service cron stop`
+
+3.  stop public-facing services, like nginx, or the service the health check is looking for ex)  `sudo service nginx stop`
+
+#### Begin data migration:
+
+1.  `ssh -A new-hostname` into new host to enable forwarding of ssh-agent
+
+2.  stop cron jobs, `sudo service cron stop`
+
+3.  stop public-facing services, like nginx, or the service the health check is looking for ex)  `sudo service nginx stop`
+
+4.  ensure that the volume is mounted in the correct location 
+
+5.  run rsync once to move bulk of data and as necessary to watch for changes, and one final time before making the swap `sudo -E -s rsync -avz --rsync-path="sudo rsync" username@hostname: /pathname/ /pathname/` 
```

Comment:
This step could probably be broken up in to a  couple steps:

- Bulk data transfer
- Check data transfer (see how many changes there are)
- Shutdown cron/services on old host
- Final data transfer